bufferevent_sock.c

```diff
@@ -431,23 +431,24 @@ bufferevent_socket_connect(struct bufferevent *bev,
 			result = 0;
 			goto done;
 		}
-	} else if (r == 1) {
+	} else {
 		/* The connect succeeded already. How very BSD of it. */
 		result = 0;
 		bufev_p->connecting = 1;
 		bufferevent_trigger_nolock_(bev, EV_WRITE, BEV_OPT_DEFER_CALLBACKS);
-	} else {
-		/* The connect failed already.  How very BSD of it. */
-		result = 0;
-		bufferevent_run_eventcb_(bev, BEV_EVENT_ERROR, BEV_OPT_DEFER_CALLBACKS);
-		bufferevent_disable(bev, EV_WRITE|EV_READ);
```

Comment:
Yes, this code is unreachable, but the intention was not to call callback if the bufferevent_socket_connect() returns error (-1), and an error will be returned in this case

Reply:
Oh, I didn't realise that this is by design.
Thanks for the review. I left only the relevant changes and renamed the PR to reflect the actual issue.